Fails to build on my review vm

<details><summary>Build log</summary>

```
lua-language-server> 测试[tclient]...
lua-language-server> test/tclient/tests/load-relative-library.lua:54: assertion failed!
lua-language-server> stack traceback:
lua-language-server>    [C]: in function 'assert'
lua-language-server>    test/tclient/tests/load-relative-library.lua:54: in upvalue 'callback'
lua-language-server>    script/lclient.lua:129: in function <script/lclient.lua:128>
lua-language-server> stack traceback:
lua-language-server>    script/lclient.lua:123: in function 'await.errorHandle'
lua-language-server>    script/await.lua:35: in function 'await.checkResult'
lua-language-server>    (...tail calls...)
lua-language-server>    script/lclient.lua:201: in local 'callback'
lua-language-server>    script/lclient.lua:232: in method 'update'
lua-language-server>    script/lclient.lua:143: in method 'start'
lua-language-server>    test/tclient/tests/load-relative-library.lua:13: in main chunk
lua-language-server>    [C]: in upvalue 'originRequire'
lua-language-server>    test.lua:42: in function 'require'
lua-language-server>    test/tclient/init.lua:11: in main chunk
lua-language-server>    [C]: in upvalue 'originRequire'
lua-language-server>    test.lua:42: in function 'require'
lua-language-server>    test.lua:48: in upvalue 'test'
lua-language-server>    test.lua:108: in local 'main'
lua-language-server>    test.lua:115: in main chunk
lua-language-server>    (bootstrap.lua):85: in main chunk
lua-language-server>    [C]: in ?
lua-language-server> bin/lua-language-server: script/lclient.lua:124: [19:47:46.386][error][#0:script/lclient.lua:123]: test/tclient/tests/load-relative-library.lua:54: assertion failed!
lua-language-server> stack traceback:
lua-language-server>    [C]: in function 'assert'
lua-language-server>    test/tclient/tests/load-relative-library.lua:54: in upvalue 'callback'
lua-language-server>    script/lclient.lua:129: in function <script/lclient.lua:128>
lua-language-server> stack traceback:
lua-language-server>    script/lclient.lua:123: in function 'await.errorHandle'
lua-language-server>    script/await.lua:35: in function 'await.checkResult'
lua-language-server>    (...tail calls...)
lua-language-server>    script/lclient.lua:201: in local 'callback'
lua-language-server>    script/lclient.lua:232: in method 'update'
lua-language-server>    script/lclient.lua:143: in method 'start'
lua-language-server>    test/tclient/tests/load-relative-library.lua:13: in main chunk
lua-language-server>    [C]: in upvalue 'originRequire'
lua-language-server>    test.lua:42: in function 'require'
lua-language-server>    test/tclient/init.lua:11: in main chunk
lua-language-server>    [C]: in upvalue 'originRequire'
lua-language-server>    test.lua:42: in function 'require'
lua-language-server>    test.lua:48: in upvalue 'test'
lua-language-server>    test.lua:108: in local 'main'
lua-language-server>    test.lua:115: in main chunk
lua-language-server>    (bootstrap.lua):85: in main chunk
lua-language-server>    [C]: in ?
lua-language-server> 
lua-language-server> stack traceback:
lua-language-server> 	[C]: in function 'error'
lua-language-server> 	script/lclient.lua:124: in function 'await.errorHandle'
lua-language-server> 	script/await.lua:35: in function 'await.checkResult'
lua-language-server> 	(...tail calls...)
lua-language-server> 	script/lclient.lua:201: in local 'callback'
lua-language-server> 	script/lclient.lua:232: in method 'update'
lua-language-server> 	script/lclient.lua:143: in method 'start'
lua-language-server> 	test/tclient/tests/load-relative-library.lua:13: in main chunk
lua-language-server> 	[C]: in upvalue 'originRequire'
lua-language-server> 	test.lua:42: in function 'require'
lua-language-server> 	test/tclient/init.lua:11: in main chunk
lua-language-server> 	[C]: in upvalue 'originRequire'
lua-language-server> 	test.lua:42: in function 'require'
lua-language-server> 	test.lua:48: in upvalue 'test'
lua-language-server> 	test.lua:108: in local 'main'
lua-language-server> 	test.lua:115: in main chunk
lua-language-server> 	(bootstrap.lua):85: in main chunk
lua-language-server> 	[C]: in ?
lua-language-server> FAILED: build/_/unit-test 
lua-language-server> bin/lua-language-server test.lua
lua-language-server> ninja: build stopped: subcommand failed.
```

</details>

### `x86_64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>lua52Packages.llscheck</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

--- a/pkgs/development/tools/language-servers/lua-language-server/default.nix
+++ b/pkgs/development/tools/language-servers/lua-language-server/default.nix
@@ -39,6 +39,10 @@ stdenv.mkDerivation (finalAttrs: {
       # this feature is not used in lua-language-server
       sed -i /filewatch/d 3rd/bee.lua/test/test.lua
 
+      # flaky tests on linux
+      # https://github.com/LuaLS/lua-language-server/issues/2926
+      sed -i /load-relative-library/d test/tclient/init.lua
+
       pushd 3rd/luamake
     ''
     + lib.optionalString stdenv.hostPlatform.isDarwin ''

